src/main/java/hudson/plugins/clover/CloverBuildAction.java

```diff
@@ -368,7 +378,25 @@ public Collection<? extends Action> getProjectActions() {
 
     private static final Logger logger = Logger.getLogger(CloverBuildAction.class.getName());
 
+    public static CloverBuildAction load(String workspacePath, ProjectCoverage result, String reportId, CoverageTarget healthyTarget, CoverageTarget unhealthyTarget) {
+        return new CloverBuildAction(workspacePath, result, reportId, healthyTarget, unhealthyTarget);
+    }
+
+    // Backward compatibility method
     public static CloverBuildAction load(String workspacePath, ProjectCoverage result, CoverageTarget healthyTarget, CoverageTarget unhealthyTarget) {
-        return new CloverBuildAction(workspacePath, result, healthyTarget, unhealthyTarget);
+        return new CloverBuildAction(workspacePath, result, null, healthyTarget, unhealthyTarget);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) return true;
+        if (obj == null || getClass() != obj.getClass()) return false;
+        CloverBuildAction that = (CloverBuildAction) obj;
+        return java.util.Objects.equals(reportId, that.reportId);
```

Comment:
The implementation of equals verifies only the single `reportId` field. This makes it usable for a pretty narrow case, when developer wants to de-duplicate instances based on the reportId only. 

While this is not a part of the [Object#equals contract](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/Object.html#equals(java.lang.Object)), it's rather a good practice to compare all significant fields in the object for equality. In this case, it's not only reportId, but also heatlhyTarget, unhealthyTarget and buildBaseDir.